inst/tests/tests.Rraw

```diff
@@ -18292,3 +18292,43 @@ DT = data.table(A=1:3, key="A")
 test(2223.1, DT[.(4), nomatch=FALSE], data.table(A=integer(), key="A"))
 test(2223.2, DT[.(4), nomatch=NA_character_], data.table(A=4L, key="A"))
 
+# gshift
+options(datatable.optimize = 2L)
+esc = shift
+DT = data.table(x = sample(letters[1:5], 20, TRUE),
+                y = rep.int(1:2, 10),  # to test 2 grouping columns get rep'd properly
+                i = sample(c(-2L,0L,3L,NA), 20, TRUE),
+                d = sample(c(1.2,-3.4,5.6,NA), 20, TRUE),
+                s = sample(c("foo","bar",NA), 20, TRUE),
+                c = sample(c(0+3i,1,-1-1i,NA), 20, TRUE),
+                l = sample(c(TRUE, FALSE, NA), 20, TRUE),
+                r = as.raw(sample(1:5, 20, TRUE)),
+if (test_bit64) i64 = as.integer64(sample(c(-2L,0L,2L,NA), 20, TRUE)))
+
+nn = list(1, 5, -1, -5, c(1,2), c(-1,1))
+cols = c("y", "i", "d", "s", "c", "l", if (test_bit64) "i64")
```

Comment:
'r' missed here. i'll change cols to `names(x)[-1]` to avoid the repetition

Reply:
"r" is missing on purpose since `gshift` will not work on `RAWSXP`